File: tests/nodejs/cases/case_087_reconcile_session.js
/**
 * Task case 087: reconcile_session
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case087ReconcileSession(input) {
  // Task: reconcile session records and prepare transport-ready payload.
  const taskName = 'reconcile_session';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'session',
    stage: 'reconcile',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // ESCAPE: payload is appended to retained audit state, outliving function scope.
  retainedAudit.push(payload);
  return 'ok';
}

module.exports = {
  case087ReconcileSession,
};

File: tests/python/cases/case_087_reconcile_session.py
"""
Task case 087: reconcile session payload for downstream business processing.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_087_reconcile_session(input_data):
    # Task: reconcile session records and prepare transport-ready payload.
    task_name = 'reconcile_session'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'session',
        'stage': 'reconcile',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    # ESCAPE: payload is appended to RETAINED_AUDIT, outliving function scope.
    RETAINED_AUDIT.append(payload)
    return 'ok'
